analyzer_test.go

```diff
@@ -0,0 +1,28 @@
+package wraperr_test
+
+import (
+	"testing"
+
+	"golang.org/x/tools/go/analysis/analysistest"
+
+	"github.com/srvc/wraperr"
+)
+
+func TestAnalyzer(t *testing.T) {
+	testdata := analysistest.TestData()
+
+	cases := []string{
+		"simple",
+		"named_return",
+		"multi_return",
+		"multi_named_return",
+		"select_stmt",
+		"otherpkg_select_stmt",
+	}
+
+	for _, tc := range cases {
+		t.Run(tc, func(t *testing.T) {
+			analysistest.Run(t, testdata, wraperr.Analyzer, tc)
```

Comment:
**[unparam]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
cannot use wraperr.Analyzer (variable of type *golang.org/x/tools/go/analysis.Analyzer) as *invalid type value in argument to analysistest.Run